app/app.go

```diff
@@ -573,6 +593,15 @@ func NewStrideApp(
 	)
 	app.ContractKeeper = wasmkeeper.NewDefaultPermissionKeeper(app.WasmKeeper)
 
+	app.WasmClientKeeper = ibcwasmkeeper.NewKeeperWithVM(
+		appCodec,
+		keys[ibcwasmtypes.StoreKey],
+		app.IBCKeeper.ClientKeeper,
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		wasmer,
+		app.GRPCQueryRouter(),
+	)
```

Comment:
What's the difference between [what Osmosis did](https://github.com/osmosis-labs/osmosis/pull/7503/files#diff-1ac782d2fab1d48ad7f7ef06cf2d58b37c99b6b4c98879a5ad86a985b73fad58R278-R286) and what you did here?

Reply:
> keys[ibcwasmtypes.StoreKey] / appKeepers.keys[ibcwasmtypes.StoreKey]

Osmosis just uses a different abstraction to manage keys/keepers/etc

> wasmer / ibcWasmConfig

I think the main difference here is they used a separate VM for ibc wasm (compared to x/wasm). [[docs](https://github.com/cosmos/ibc-go/blob/main/docs/docs/03-light-clients/04-wasm/03-integration.md#keeper-instantiation)] and [[osmosis note](https://github.com/osmosis-labs/osmosis/pull/7503/files#diff-1ac782d2fab1d48ad7f7ef06cf2d58b37c99b6b4c98879a5ad86a985b73fad58R278)]

Reply:
also the IBC team just sent over this [better guide](https://ibc.cosmos.network/v7/ibc/light-clients/wasm/integration/) that's made for ibc v7

Reply:
> I think the main difference here is they used a separate VM

We do too